ProjNet/CoordinateSystems/Transformations/MathTransform.cs

```diff
@@ -1,246 +1,523 @@
+        }
+
+
+        /// <summary>
+        /// Transforms a series of 2-dimensional <see cref="XY"/>-points and and a series of z-ordinate values.
+        /// </summary>
+        /// <param name="inXys">A series of <see cref="XY"/> points</param>
+        /// <param name="inZs">A series of z-ordinate values.</param>
+        /// <param name="outXys">A buffer for <see cref="XY"/> points</param>
+        /// <param name="outZs">A buffer for z-ordinate values</param>
+        /// <exception cref="ArgumentException">If the provided series' and buffers don't match in size.</exception>
+        public void Transform(ReadOnlySpan<XY> inXys, ReadOnlySpan<double> inZs, Span<XY> outXys, Span<double> outZs)
+        {
+            if (inXys.Length != outXys.Length ||
+                (inZs.Length != 0 && inXys.Length != outZs.Length) ||
+                (outZs.Length != 0 && inXys.Length != outZs.Length))
+            {
+                throw new ArgumentException("Provided spans must be the same length.");
+            }
+            var read = MemoryMarshal.Cast<XY, double>(inXys);
+            var inXs = read.Slice(0, read.Length - 1);
+            var inYs = read.Slice(1, read.Length - 1);
+
+            var write = MemoryMarshal.Cast<XY, double>(outXys);
+            var outXs = write.Slice(0, write.Length - 1);
+            var outYs = write.Slice(1, write.Length - 1);
+
+            if (inZs.Length == 0)
+                TransformCore(inXs, inYs, _inZs, outXs, outYs, _outZs, 2,2, 0);
+            else
+                TransformCore(inXs, inYs, inZs, outXs, outYs, outZs, 2, 2, 1);
+        }
+
+        /// <summary>
+        /// Transforms a series of 3-dimensional <see cref="XYZ"/>-points.
+        /// </summary>
+        /// <param name="inXyzs">A series of <see cref="XYZ"/> points</param>
+        /// <param name="outXyzs">A buffer for <see cref="XYZ"/> points</param>
+        /// <exception cref="ArgumentException">If the provided series and buffer don't match in size.</exception>
+        public void Transform(ReadOnlySpan<XYZ> inXyzs, Span<XYZ> outXyzs)
+        {
+            if (inXyzs.Length != outXyzs.Length)
+            {
+                throw new ArgumentException("Observed spans must be the same length.");
+            }
+
+            var read = MemoryMarshal.Cast<XYZ, double>(inXyzs);
+            var inXs = read.Slice(0, read.Length - 2);
+            var inYs = read.Slice(1, read.Length - 2);
+            var inZs = read.Slice(2, read.Length - 2);
+
+            var write = MemoryMarshal.Cast<XYZ, double>(outXyzs);
+            var outXs = write.Slice(0, write.Length - 2);
+            var outYs = write.Slice(1, write.Length - 2);
+            var outZs = write.Slice(2, write.Length - 2);
+
+            TransformCore(inXs, inYs, inZs, outXs, outYs, outZs,3,3,3);
+        }
+
+        #endregion
+
+        private static SequenceCoordinateConverterBase _sequenceCoordinateConverter;
```

Comment:
This will have to move to a different project. I don't think we should add a dependency to NetTopologySuite.Core for using this library.

Neither `Proj4` nor `OSGeo.GDAL.Osr` nor `DotSpatial.Projections` require a dependency on some geometry library.

Reply:
> I don't think we should add a dependency to NetTopologySuite.Core for using this library.

Oh right, you wanted to get rid of the `ICoordinateSequence` methods in here, didn't you.

That's fine, then, we can leave this part for now if they're just going away anyway.